js_modules/dagster-ui/packages/ui-core/src/selection/useSelectionInputLintingAndHighlighting.tsx

```diff
@@ -0,0 +1,140 @@
+import {applyStaticSyntaxHighlighting} from './SelectionInputHighlighter';
+import {useUpdatingRef} from '../hooks/useUpdatingRef';
+
+export const useSelectionInputLintingAndHighlighting = ({
+  cmInstance,
+  value,
+  linter,
+}: {
+  cmInstance: React.MutableRefObject<CodeMirror.Editor | null>;
+  value: string;
+  linter: (content: string) => SyntaxError[];
+}) => {
+  const errors = useMemo(() => {
+    const errors = linter(value);
+    return errors.map((error, idx) => ({
+      ...error,
+      idx,
+    }));
+  }, [linter, value]);
+
+  useLayoutEffect(() => {
+    if (!cmInstance.current) {
+      return;
+    }
+    applyStaticSyntaxHighlighting(cmInstance.current, errors);
+  }, [cmInstance, errors]);
+
+  const errorsRef = useUpdatingRef(errors);
+
+  const [error, setError] = useState<{
+    error: SyntaxError;
+    x: number;
+    y: number;
+  } | null>(null);
+
+  const errorRef = useUpdatingRef(error);
+
+  useLayoutEffect(() => {
+    const listener = (ev: MouseEvent) => {
+      if (!(ev.target instanceof HTMLElement)) {
+        return;
+      }
+      const error = ev.target.closest('.selection-input-error') as HTMLElement | null;
+      if (error) {
+        const regex = /selection-input-error-(\d+)/;
+        const errorIdx = parseInt(error.className.match(regex)?.[1] ?? '0', 10);
+        const errorAnnotation = errorsRef.current[errorIdx];
+        if (errorAnnotation) {
+          setError({
+            error: errorAnnotation,
+            x: ev.clientX,
+            y: ev.clientY,
+          });
+          return;
+        }
+      }
+      if (errorRef.current) {
+        setError(null);
+      }
+    };
+    document.body.addEventListener('mousemove', listener);
+    return () => {
+      document.body.removeEventListener('mousemove', listener);
+    };
+  }, [cmInstance, errorsRef, errorRef]);
+
+  const message = useMemo(() => {
+    if (!error) {
+      return null;
+    }
+    if (error.error.mismatchedInput && error.error.expectedInput) {
+      return (
+        <Box>
+          Mismatched input{' '}
+          <MonoSmall color={Colors.textRed()}>&apos;{error.error.mismatchedInput}&apos;</MonoSmall>.
+          Expected{' '}
+          {error.error.expectedInput.map((expectedInput, idx) => (
+            <>
+              {idx > 0 && ', '}
+              <MonoSmall key={expectedInput} color={Colors.textBlue()}>
+                {expectedInput}
+              </MonoSmall>
+            </>
+          ))}
+        </Box>
+      );
+    }
+    if (error.error.message) {
+      return error.error.message;
+    }
+    return null;
+  }, [error]);
+
+  if (!error) {
+    return null;
+  }
+
+  return ReactDOM.createPortal(
+    <PortalElement $bottom={error.y} $left={error.x}>
+      <Box
+        as={Content}
+        padding={{horizontal: 12, vertical: 8}}
+        flex={{direction: 'row', gap: 4}}
+        color={Colors.textLight()}
+      >
+        <Icon name="run_failed" color={Colors.accentRed()} />
```

Comment:
`error_outline` might look a little less loud and a little cleaner than the filled-in error icon.